specification/communication/data-plane/Microsoft.CommunicationServicesCommon/preview/2020-11-19-preview1/common.json

```diff
@@ -0,0 +1,114 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "Communication",
+    "description": "Communication Services API common types.",
+    "version": "2020-11-19-preview1"
+  },
+  "paths": {},
+  "definitions": {
+    "CommunicationError": {
+      "description": "The Communication Services error exception.",
+      "required": [
+        "error"
+      ],
+      "properties": {
+        "error": {
+          "description": "The Communication Services error.",
+          "$ref": "#/definitions/Error"
+        }
+      }
+    },
+    "Error": {
+      "description": "The Communication Services error.",
+      "required": [
+        "code",
+        "message"
+      ],
+      "properties": {
+        "code": {
+          "type": "string",
+          "description": "The error code."
+        },
+        "message": {
+          "type": "string",
+          "description": "The error message."
+        },
+        "target": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The error target."
+        },
+        "details": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The error details."
+        },
+        "innererror": {
```

Comment:
can there be more than a single inner error? Perhaps it would be better to make this an array.

Reply:
I've followed the schema definition from the guidelines, innererror is a single nested error object.

That said I realized that `details` should be an array of Errors, so I've changed that.

https://github.com/microsoft/api-guidelines/blob/vNext/Guidelines.md#7102-error-condition-responses